Shell/main.cpp

```diff
@@ -223,6 +224,47 @@ struct CommandTimer {
     CElapsedTimer timer;
 };
 
+static Vector<String> process_arguments(const Vector<String>& args)
+{
+    Vector<String> argv_string;
+    for (auto& arg : args) {
+        bool is_glob = false;
+        for (int i = 0; i < arg.length(); i++) {
+            char c = arg.characters()[i];
+            if (c == '*' || c == '?') {
+                is_glob = true;
+            }
+        }
+
+        if (is_glob == false) {
+            argv_string.append(arg.characters());
+        } else {
+            CDirIterator di(".", CDirIterator::NoFlags);
+            if (di.has_error()) {
+                fprintf(stderr, "CDirIterator: %s\n", di.error_string());
+                continue;
+            }
+
+            while (di.has_next()) {
+                String name = di.next_path();
+
+                // Dotfiles have to be explicitly requested
+                if (name[0] == '.' && arg[0] != '.')
+                    continue;
+
+                // And even if they are, skip . and ..
+                if (name == "."  || name == "..")
```

Comment:
bash includes "." and ".." in "echo .*"
IIRC you were using zsh, are you getting a different result there?

Reply:
Yes indeed. Filtered out here :)

I think that filtering them is a nice thing to do unless we absolutely have to include them. They are really rather useless.